src/NSwag.CodeGeneration/Models/ResponseModelBase.cs

```diff
@@ -85,9 +85,16 @@ public bool IsDate
         }
 
         /// <summary>Gets a value indicating whether the response requires a text/plain content.</summary>
-        public bool IsPlainText =>
-            !_response.Content.ContainsKey("application/json") &&
-            (_response.Content.ContainsKey("text/plain") || _operationModel.Produces == "text/plain");
+        public bool IsPlainText
+        {
+            get
+            {
+                JsonObjectType primitive = JsonObjectType.String | JsonObjectType.Integer | JsonObjectType.Number | JsonObjectType.Boolean;
+                return ActualResponseSchema != null &&
+                    (ActualResponseSchema.Type & primitive) == ActualResponseSchema.Type &&
+                    (ActualResponseSchema.Type | primitive) == primitive;
```

Comment:
If the response schema only contains primitive data types, then the response can be considered plain text.